backend/endpoints/main.go

```diff
@@ -65,7 +67,13 @@ func (fn handler[T, V]) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	// construct a dependency factory for this request, which implies instantiating a logger
 	logger := buildLogger(r.Method, r.URL.Path)
-	dependencyFactory := DependencyProvider{Log: logger}
+	var frontendid int
+	ctx := contexts.GetDatabaseContext()
+	err := ctx.Query("SELECT frontendid from frontend where frontendurl = $1;", []interface{}{r.URL.Path}, &frontendid)
+	if err != nil {
+		log.Println("frontend doesn't exist", err.Error())
+	}
```

Comment:
I don't like the idea of querying the database directly from the endpoint package, this should be in the repositories package with its own repository type, IE:

```go
type (
    IFrontendRepository interface {
        GetFrontendFromURL(url string) int
    }
)
```

We can then acquire and call the repository as follows:
```go
frontendRepo := repositories.GetRepository(FRONTEND).(IFrontendRepository)
frontendID := frontendRepo.GetFrontendFromURL(r.URL.Path)
```

Reply:
Also the logger being used is incorrect, endpoints have their own custom logger type so we can easilly group together relevant log messages without them being tangled together in a mess, please use the logger defined on line 69